segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/StorageHealthTrackerTests.java

```diff
@@ -0,0 +1,882 @@
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@Slf4j
+public class StorageHealthTrackerTests extends ThreadPooledTestSuite {
+
+    public static final int CONTAINER_ID = 10;
+
+    @Test
+    public void testDefault() {
+        val delaySupplier = spy(new TestDelaySupplier());
+        val storageHealthTracker = new StorageHealthTracker(CONTAINER_ID, ChunkedSegmentStorageConfig.DEFAULT_CONFIG,
+                System::currentTimeMillis,
+                delaySupplier);
+
+        checkState(storageHealthTracker, false, false, false, 0);
+
+        // Start iteration
+        storageHealthTracker.beginIteration(0);
+        checkState(storageHealthTracker, false, false, false, 0);
+
+        // Call throttle
+        checkThrottling(storageHealthTracker, delaySupplier, 0);
+        checkState(storageHealthTracker, false, false, false, 0);
+
+        // End iteration
+        storageHealthTracker.endIteration(0);
+        checkState(storageHealthTracker, false, false, false, 0);
+    }
+
+    private void checkState(StorageHealthTracker storageHealthTracker, boolean expectedStorageUnavailable, boolean expectedStorageFull, boolean expectedSafeMode, int expectedPercentLate) {
+        Assert.assertEquals(expectedStorageUnavailable, storageHealthTracker.isStorageUnavailable());
+        Assert.assertEquals(expectedStorageFull, storageHealthTracker.isStorageFull());
+        Assert.assertEquals(expectedSafeMode, storageHealthTracker.isSafeMode());
+        Assert.assertEquals(expectedPercentLate, storageHealthTracker.getPercentLate());
+    }
+
+    @Test
+    public void testDefaultNormal() {
+        val delaySupplier = spy(new TestDelaySupplier());
+        val storageHealthTracker = new StorageHealthTracker(CONTAINER_ID, ChunkedSegmentStorageConfig.DEFAULT_CONFIG,
+                System::currentTimeMillis,
+                delaySupplier);
+
+        checkState(storageHealthTracker, false, false, false, 0);
+        for (int n = 0; n < 3; n++) {
+            // Start iteration
+            storageHealthTracker.beginIteration(n);
+            checkState(storageHealthTracker, false, false, false, 0);
+
+            // Call throttle
+            checkThrottling(storageHealthTracker, delaySupplier, 0);
+
+            checkState(storageHealthTracker, false, false, false, 0);
+            TestUtils.addRequestStats(storageHealthTracker, 0, 5, 0, 0);
+            // End iteration
+            storageHealthTracker.endIteration(n);
+            checkState(storageHealthTracker, false, false, false, 0);
+        }
+    }
+
+    @Test
+    public void testTransition_Normal_Slow_Normal() {
```

Comment:
Please, user camel case for method names.

Reply:
I know . I had camel case but it was difficult to keep to keep track of transitions. I'll revert back to camelCase

Reply:
fixed